File: library/src/test/java/com/basgeekball/awesomevalidation/ValidationStyleTest.java
package com.basgeekball.awesomevalidation;

import junit.framework.TestCase;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;

@RunWith(JUnit4.class)
public class ValidationStyleTest extends TestCase {

    public void testFromValueToBasic() {
        assertEquals(ValidationStyle.BASIC, ValidationStyle.fromValue(0));
    }

    public void testFromValueToColoration() {
        assertEquals(ValidationStyle.COLORATION, ValidationStyle.fromValue(1));
    }

    public void testFromValueToUnderlabel() {
        assertEquals(ValidationStyle.UNDERLABEL, ValidationStyle.fromValue(2));
    }

    public void testFromValueToTextInputLayout() {
        assertEquals(ValidationStyle.TEXT_INPUT_LAYOUT, ValidationStyle.fromValue(3));
    }

    @Test(expected = IllegalArgumentException.class)
    public void testFromValueThrowException() {
        ValidationStyle.fromValue(4);
    }

}

File: library/src/test/java/com/basgeekball/awesomevalidation/ValidationStyleTest.kt
package com.basgeekball.awesomevalidation

import junit.framework.TestCase

import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.JUnit4

@RunWith(JUnit4::class)
class ValidationStyleTest : TestCase() {

    @Test
    fun testFromValueToBasic() {
        assertEquals(ValidationStyle.BASIC, ValidationStyle.fromValue(0))
    }

    @Test
    fun testFromValueToColoration() {
        assertEquals(ValidationStyle.COLORATION, ValidationStyle.fromValue(1))
    }

    @Test
    fun testFromValueToUnderlabel() {
        assertEquals(ValidationStyle.UNDERLABEL, ValidationStyle.fromValue(2))
    }

    @Test
    fun testFromValueToTextInputLayout() {
        assertEquals(ValidationStyle.TEXT_INPUT_LAYOUT, ValidationStyle.fromValue(3))
    }

    @Test(expected = IllegalArgumentException::class)
    fun testFromValueThrowException() {
        ValidationStyle.fromValue(4)
    }
}
